packages/next/src/server/next-server.ts

```diff
@@ -205,6 +206,24 @@ export default class NextNodeServer extends BaseServer {
       this.imageResponseCache = new ResponseCache(this.minimalMode)
     }
 
+    if (!options.dev) {
```

Comment:
```suggestion
    if (!options.dev && this.nextConfig.experimental.instrumentationHook) {
```
Should this be gated behind the config?

Reply:
should be fine because the compilation of the file is gated so it should just noop